### ISSUE TITLE
pickle dump leaks memory

### ISSUE THREAD

**Issue by mayankasthana** [ascv/HyperLogLog#33]
### Reproduction:

```
from HLL import HyperLogLog
import pickle
import os
import psutil

process = psutil.Process(os.getpid())
def print_memory_usage():
    B = process.memory_info().rss
    MB = B / (1<<20)
    print(f'{MB} MB')


for i in range(10000):
    hll = HyperLogLog(14)
    pickle.dumps(hll)
    if i % 1000 == 0:
        print_memory_usage()

```

### Output

```
46.69140625 MB
78.109375 MB
109.5625 MB
140.7578125 MB
172.32421875 MB
203.77734375 MB
235.23046875 MB
266.42578125 MB
297.87890625 MB
329.33203125 MB
```

Without `pickle.dumps`

```
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
46.63671875 MB
```

### Environment
Python version: 3.7.3

**Comment by mayankasthana** [comment]
I did check with the fixes from https://github.com/ascv/HyperLogLog/pull/31/commits/00e97fdc1719a4db63aea77b29f734b60c5cf1c3  (#31) but the memory leak is still present.

**Comment by ascv** [comment]
What version of python were you using to test?

**Comment by ascv** [comment]
It's possible [#31] does resolve a memory leak but pickle itself is also a problem in this test. You can try un-importing pickle after calling `pickle.dumps` and then checking the memory usage.

**Comment by mayankasthana** [comment]
> What version of python were you using to test?

Python version: 3.7.3

**Comment by mayankasthana** [comment]
> It's possible [#31] does resolve a memory leak but pickle itself is a problem in this test. You can try un-importing pickle after calling `pickle.dumps` and then checking the memory usage.

Not sure how I can unimport pickle. I tried doing a reload - 
```
import importlib
for i in range(10000):
    importlib.reload(pickle)
    hll = HyperLogLog(14)
    pickle.dumps(hll)
    if i % 1000 == 0:
        print_memory_usage()
```

**Comment by mayankasthana** [comment]
Meanwhile, exporting the registers and recreating the HLL using `set_registers` doesn't leak memory. So I am using that.

**Comment by ascv** [comment]
It turns out that both `args` and `registers` need to have their reference count manually decremented since `Py_BuildValue` creates its own copies when building the tuple. I commented on [#31] with the requested changes.

**Comment by ascv** [comment]
Thanks for contributing to this project. This will be live in PyPy in a few days. I'm bundling the next distribution with [#31] alongside some other changes.

**Comment by mayankasthana** [comment]
Thanks a lot @ascv and @mauguignard to get this fixed so quickly.

**Comment by ascv** [comment]
Thanks @mauguignard for contributing a fix [#31] and to @mayankasthana for testing this issue and contributing [#35]. I got the usernames mixed up in my previous comment :)
